### PR TITLE
fix data race in client and daemon (#2168)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #2168 to release-2.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In chaos-mesh repo:
git pr https://github.com/chaos-mesh/chaos-mesh/pull/2175
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/chaos-mesh.git pr/2175:release-2.0-d0f184c38a4c
```

---

Signed-off-by: YangKeao <keao.yang@yahoo.com>

### What problem does this PR solve?

Fix two possible data race and add an `UpdatedStatusClient`:

1. The `blockingBuffer` is full of data race while reading and writing at the same time.
2. The object returned from the `cache` could be the same one, and the reference stored in it will be simply copied out. Run `DeepCopy` before assigning it to the new one.

### CODE REVIEW

**Review by fewdan — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -71,6 +71,7 @@ require (
 	github.com/syndtr/gocapability v0.0.0-20180916011248-d98352740cb2 // indirect
 	github.com/vishvananda/netlink v1.0.0
 	github.com/vishvananda/netns v0.0.0-20171111001504-be1fbeda1936 // indirect
+	go.uber.org/atomic v1.6.0
```
[the remark this answers is not in the document]
Comment (STRRL) [reply go.mod:74]: @YangKeao how about replacing it with a newer version? I noticed that the latest version is v1.9.0

https://github.com/uber-go/atomic/releases

[the remark this answers is not in the document]
Comment (STRRL) [reply go.mod:74]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (STRRL) [reply go.mod:74]: what the hell? I can not @ this bot.

[the remark this answers is not in the document]
Comment (STRRL) [reply go.mod:74]: ignore @muse-dev

[the remark this answers is not in the document]
Comment (YangKeao) [reply go.mod:74]: ignore

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**